[5.1] Use tinyint(1) for boolean type definition

df-arif opened the pull request:
Using tinyint(1) instead of just tinyint for SQLite boolean type works better. SQLite doesn't natively support boolean data type. A tinyint field without a size could be an integer but when using tinyint(1) then it's much safer to treat this as boolean.

Currently when I am running my app using MySql database I can easily identify my boolean fields with the type tinyint(1). However, when running my app using SQLite the same field shows up with type tinyint. So, basically getting different results when switching databases.

df-arif committed:
Using tinyint(1) for boolean type definition.

Using tinyint(1) instead of just tinyint for SQLite boolean type works better.

GrahamCampbell commented:
Please see the other rejections.

GrahamCampbell commented:
Oh wait. Missread this.

GrahamCampbell commented:
Please update the test case.

```
There was 1 failure:
1) DatabaseSQLiteSchemaGrammarTest::testAddingBoolean
Failed asserting that two strings are equal.
--- Expected
+++ Actual
@@ @@
-'alter table "users" add column "foo" tinyint not null'
+'alter table "users" add column "foo" tinyint(1) not null'
/home/travis/build/laravel/framework/tests/Database/DatabaseSQLiteSchemaGrammarTest.php:327
```

df-arif committed:
Updating test case for using tinyint(1) as boolean

Updating test case for using tinyint(1) as field definition for SQLite boolean type.
